### PR TITLE
Fix CPU-only runtime for dpcpp-generated executables in custom easyblock for intel-compilers

### PR DESCRIPTION
**Pull request by bartoldeman**
First of all the OpenCL runtime needs to be aware of the CPU-only shared object via `OCL_ICD_FILENAMES -> libintelocl.so`
The module also needs to set TBB-related variables (the same ones as the old iccifort modules), since the above object links to TBB.

This can be tested using e.g. saxpy from
https://github.com/jeffhammond/dpcpp-tutorial

I haven't tested dpcpp on FPGAs, nor on Intel GPUs (though I can try the one on my laptop, an Intel Corporation UHD Graphics 620 chip).

NVidia GPUs are not supported.

### CODE REVIEW

File: easybuild/easyblocks/i/intel_compilers.py [hunk 1]
```diff
@@ -120,12 +121,20 @@ def make_module_req_guess(self):
             os.path.join('compiler', 'lib', 'intel64_lin'),
         ]
         libdirs = [os.path.join(self.compilers_subdir, x) for x in libdirs]
+        libdirs.append(os.path.join(self.tbb_subdir, 'lib', 'intel64', 'gcc4.8'))
```
Comment (akesandgren) [inline easybuild/easyblocks/i/intel_compilers.py:124]: This should use globbing since gcc4.8 may be changed.

**Review by akesandgren — DISMISSED** [review]

File: easybuild/easyblocks/i/intel_compilers.py (hunk 1 above)
Comment (boegel) [reply easybuild/easyblocks/i/intel_compilers.py:124]: @bartoldeman +1 on this

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _intel-compilers-2021.1.2.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
build-node.computecanada.ca - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.7.7
See https://gist.github.com/8d15624e6d4528d9527c309cdd2ec2c6 for a full test report.

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _intel-compilers-2021.2.0.eb_ (partial log available at https://gist.github.com/d8f5a93a8d1e9cba49754471bb394617)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
build-node.computecanada.ca - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.7.7
See https://gist.github.com/a0f2873cac497bff6ec47e3a8e46fe10 for a full test report.

File: easybuild/easyblocks/i/intel_compilers.py (hunk 1 above)
Comment (bartoldeman) [reply easybuild/easyblocks/i/intel_compilers.py]: Used the get_tbb_gccprefix function for this now, like in the icc easyblock.

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _intel-compilers-2021.2.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
build-node.computecanada.ca - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.7.7
See https://gist.github.com/f7312e80f6b5f11e77bb6ab4dbf9868f for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _intel-compilers-2021.1.2.eb_ 
 * **SUCCESS** _intel-compilers-2021.2.0.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3599.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/d672a744689b48b396725345a75d9488 for a full test report.

**Pull request merged**